test/nbrowser/DocUsageTracking.ts

```diff
@@ -0,0 +1,179 @@
+import {UserAPI} from 'app/common/UserAPI';
+import {assert, driver, Key} from 'mocha-webdriver';
+import fetch from 'node-fetch';
+import * as gu from 'test/nbrowser/gristUtils';
+import {server} from 'test/nbrowser/testServer';
+import {setupTestSuite} from 'test/nbrowser/testUtils';
+import {
+  enableExternalAttachmentsForTestSuite
+} from 'test/nbrowser/externalAttachmentsHelpers';
+
+describe('DocumentUsage', function() {
+  this.timeout(20000);
+  const cleanup = setupTestSuite();
+
+  const ownerUser = 'user1';
+  let api: UserAPI;
+  let session: gu.Session;
+
+  enableExternalAttachmentsForTestSuite();
+
+  async function makeSessionAndLogin() {
+    session = await gu.session().user(ownerUser).login();
+    api = session.createHomeApi();
+  }
+
+  before(async function () {
+    await makeSessionAndLogin();
+  });
+
+  it('shows usage stats on the raw data page', async function() {
+    await session.tempNewDoc(cleanup, "EmptyUsageDoc");
+
+    // Check that the Usage section exists.
+    await goToDocUsage();
+    assert.equal(await driver.find('.test-doc-usage-heading').getText(), 'Usage');
+    await assertUsageMessage(null);
+
+    // Check that usage is at 0.
+    await assertRowCount('0');
+    await assertDataSize('0.00');
+    await assertAttachmentsSize('0.00');
+
+    // Check that banners aren't shown on the raw data page.
+    await gu.assertBannerText(null);
+  });
+
+  function testAttachmentsUsage(getDocId: () => string) {
+    it('updates attachments size usage when uploading attachments', async function () {
+      const docId = getDocId();
+      // Add a new 'Attachments' column of type Attachment to Table1.
+      await api.applyUserActions(docId, [['AddEmptyTable', "AttachmentsTable"]]);
+      await gu.getPageItem('AttachmentsTable').click();
+      await gu.waitForServer();
+      await addAttachmentColumn('Attachments');
```

Comment:
```suggestion
      await gu.sendActions([['AddEmptyTable', "AttachmentsTable"]]);
      await gu.getPageItem('AttachmentsTable').click();
      await gu.addColumn('Attachments', 'Attachment');
```

When you change the document through the API, you are not waiting for those change to propagte back to browser. It works 99% of time, but introduce flakines.

The second `waitForServer` is not needed I think, is it?

Your helper for adding attachments column is doing the same thing that this call:
`await gu.addColumn('Attachments', 'Attachment');`

Reply:
Made these changes, you're 100% right, that second waitForServer isn't needed - gu.sendActions waits :)